pkgs/applications/networking/cluster/jx/default.nix

```diff
@@ -1,49 +1,49 @@
-{ stdenv, buildGoModule, fetchFromGitHub, lib, installShellFiles }:
+{ stdenv, buildGoModule, fetchFromGitHub, lib, nix-update-script, go }:
 
 buildGoModule rec {
   pname = "jx";
-  version = "2.1.155";
+  version = "3.10.144";
 
   src = fetchFromGitHub {
     owner = "jenkins-x";
     repo = "jx";
     rev = "v${version}";
-    sha256 = "sha256-kwcmZSOA26XuSgNSHitGaMohalnLobabXf4z3ybSJtk=";
+    sha256 = "sha256-1/gW9p2M0ZW3H9rM7JNdsh19A4yIO8SCE8u6UpB4ftI=";
   };
 
-  vendorHash = "sha256-ZtcCBXcJXX9ThzY6T0MhNfDDzRC9PYzRB1VyS4LLXLs=";
+  vendorHash = "sha256-AIaZVkWdNj1Vsrv2k4B5lLE0lOFuiTD7lwS/DikmC14=";
 
-  doCheck = false;
+  subPackages = [ "cmd" ];
 
-  subPackages = [ "cmd/jx" ];
-
-  nativeBuildInputs = [ installShellFiles ];
+  CGO_ENABLED = 0;
+  env.COSIGN_EXPERIMENTAL = "true";
 
   ldflags = [
-    "-s -w"
-    "-X github.com/jenkins-x/jx/pkg/version.Version=${version}"
-    "-X github.com/jenkins-x/jx/pkg/version.Revision=${src.rev}"
-    "-X github.com/jenkins-x/jx/pkg/version.GitTreeState=clean"
+    "-s"
```

Comment:
You probably want to remove this flag as well, see #274503

Reply:
I'm aware of these discussions. I'm a fan of the Go stripping, and I think ideally it would be set by `buildGoModule` automatically and just be dropped when `dontStrip = true`. I don't know much about what the nix stripping does/should do, but it isn't as good. For this concrete example:

With `-s`:

```
❯ file result/bin/jx 
result/bin/jx: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), statically linked, stripped

❯ l result/bin/jx 
Permissions Size User Date Modified Name
.r-xr-xr-x   36M root  1 Jan  1970  result/bin/jx
```

Without `-s`:

```
❯ file result/bin/jx 
result/bin/jx: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), statically linked, not stripped

Permissions Size User Date Modified Name
.r-xr-xr-x   41M root  1 Jan  1970  result/bin/jx
```

So I'd like to keep that flag for now.

Reply:
Fine by me.